packages/editor/src/components/rich-text/index.js

```diff
@@ -444,22 +444,38 @@ export class RichText extends Component {
 				return;
 			}
 
+			// These modifiers would not result in modification of text content,
+			// if they're pressed they should not cause merge or deletion of blocks.
+			// Returning here also allows further binding of these events.
+			const isMetaShiftDeletePressed = event.metaKey && event.shiftKey && keyCode === DELETE;
+			const isCtrlShiftDeletePressed = event.ctrlKey && event.shiftKey && keyCode === DELETE;
+			const isCtrlAltDeletePressed = event.ctrlKey && event.altKey && keyCode === DELETE;
+			const isMetaAltPressed = event.metaKey && event.altKey;
+
+			if ( isMetaShiftDeletePressed || isCtrlShiftDeletePressed || isCtrlAltDeletePressed || isMetaAltPressed ) {
```

Comment:
Testing on a Mac, `isCtrlShiftDeletePressed` and `isCtrlAltDeletePressed` still delete characters in apps such as Notes. Is this different on Windows? But if we want shortcuts to work for these, we could also just return if any modifier is pressed?

Reply:
Hmm, I just tested Notes, TextEdit along with input and textarea elements. Consistently across all of them on the mac Ctrl+Option+Delete and Ctrl+Shift+Delete (just delete and not backspace) doesn't result in any text deletion, and I had the same results on Windows. 

I initially thought about returning early for any modifier, but there are some things that I think we should continue to handle like Cmd+Shift+Backspace (delete preceding text to the start of the line) and Alt+Shift+Backspace (delete preceding word). That these continue to work across multiple paragraph blocks is quite a nice user experience.